src/v0/destinations/marketo_static_list/transform.js

```diff
@@ -140,7 +129,24 @@ const processRouterDest = async (inputs, reqMetadata) => {
   // If true then previous status is 500 and every subsequent event output should be
   // sent with status code 500 to the router to be retried.
   const tokenisedInputs = inputs.map((input) => ({ ...input, token }));
-  const respList = await simpleProcessRouterDest(tokenisedInputs, processEvent, reqMetadata);
+  const leadIdObj = {
+    insert: [],
+    delete: [],
+  };
+  // use lodash.groupby to group the inputs based on message type
+  const groupedInputs = lodash.groupBy(tokenisedInputs, (input) => input.message.type);
```

Comment:
What if groupedInputs have messages of a type different from record and audiencelist? What happens to those events?

Reply:
They shouldn't come through
As we have set supported messageTypes in the config to only "record" or "audienceList".
CC @utsabc

Reply:
[Here](https://github.com/rudderlabs/rudder-integrations-config/blob/e1964fbf74372761c076be00df9eedb285f036ed/src/configurations/destinations/marketo_static_list/db-config.json#L22)

Reply:
That would be an edge case where `loadash` would create another group for those events but those will remain unprrcessed as per code below